Mathlib/LinearAlgebra/AffineSpace/ContinuousAffineEquiv.lean

```diff
@@ -0,0 +1,295 @@
+
+instance equivLike : EquivLike (P₁ ≃ᵃL[k] P₂) P₁ P₂ where
+  coe f := f.toFun
+  inv f := f.invFun
+  left_inv f := f.left_inv
+  right_inv f := f.right_inv
+  coe_injective' _ _ h _ := toAffineEquiv_injective (DFunLike.coe_injective h)
+
+instance : CoeFun (P₁ ≃ᵃL[k] P₂) fun _ ↦ P₁ → P₂ :=
+  DFunLike.hasCoeToFun
+
+attribute [coe] ContinuousAffineEquiv.toAffineEquiv
+
+/-- Coerce continuous affine equivalences to affine equivalences. -/
+instance ContinuousAffineEquiv.coe : Coe (P₁ ≃ᵃL[k] P₂) (P₁ ≃ᵃ[k] P₂) := ⟨toAffineEquiv⟩
+
+theorem coe_injective : Function.Injective ((↑) : (P₁ ≃ᵃL[k] P₂) → P₁ ≃ᵃ[k] P₂) := by
+  intro e e' H
+  cases e
+  congr
+
+instance funLike : FunLike (P₁ ≃ᵃL[k] P₂) P₁ P₂ where
+  coe f := f.toAffineEquiv
+  coe_injective' _ _ h := coe_injective (DFunLike.coe_injective h)
+
+@[simp, norm_cast]
+theorem coe_coe (e : P₁ ≃ᵃL[k] P₂) : ⇑(e : P₁ ≃ᵃ[k] P₂) = e :=
+  rfl
+
+@[simp]
+theorem coe_toEquiv (e : P₁ ≃ᵃL[k] P₂) : ⇑e.toEquiv = e :=
+  rfl
+
+-- NOTE(MR): I have omitted `coe_mk`, `coe_mk'`, `coe_inj`, `coeFn_injective` lemmas for now;
+-- happy to add them!
+
+-- NOTE(MR): the next three items are cargo-culted; please review carefully if they make sense!
+/-- See Note [custom simps projection].
+  We need to specify this projection explicitly in this case,
+  because it is a composition of multiple projections. -/
+def Simps.apply (e : P₁ ≃ᵃL[k] P₂) : P₁ → P₂ :=
+  e
+
+/-- See Note [custom simps projection]. -/
+def Simps.coe (e: P₁ ≃ᵃL[k] P₂) : P₁ ≃ᵃ[k] P₂ :=
+  e
+
+initialize_simps_projections ContinuousLinearMap (toAffineEquiv_toFun → apply, toAffineEquiv → coe)
+
+@[ext]
+theorem ext {e e' : P₁ ≃ᵃL[k] P₂} (h : ∀ x, e x = e' x) : e = e' :=
+  DFunLike.ext _ _ h
+
+theorem ext_iff {e e' : P₁ ≃ᵃL[k] P₂} : e = e' ↔ ∀ x, e x = e' x :=
+  DFunLike.ext_iff
+
+@[continuity]
+protected theorem continuous (e : P₁ ≃ᵃL[k] P₂) : Continuous e :=
+  e.2
+
+end Basic
+
+section ReflSymmTrans
+
+variable (k P₁) in
+/-- Identity map as a `ContinuousAffineEquiv`. -/
+def refl : P₁ ≃ᵃL[k] P₁ where
+  toEquiv := Equiv.refl P₁
+  linear := LinearEquiv.refl k V₁
+  map_vadd' _ _ := rfl
+
+@[simp]
+theorem coe_refl : ⇑(refl k P₁) = id :=
+  rfl
+
+@[simp]
+theorem refl_apply (x : P₁) : refl k P₁ x = x :=
+  rfl
+
+@[simp]
+theorem toAffineEquiv_refl : (refl k P₁).toAffineEquiv = AffineEquiv.refl k P₁ :=
+  rfl
+
+@[simp]
+theorem toEquiv_refl : (refl k P₁).toEquiv = Equiv.refl P₁ :=
+  rfl
+
+/-- Inverse of a continuous affine equivalence as a continuous affine equivalence. -/
+@[symm]
+def symm (e : P₁ ≃ᵃL[k] P₂) : P₂ ≃ᵃL[k] P₁ where
+  toAffineEquiv := e.toAffineEquiv.symm
+  continuous_toFun := e.continuous_invFun
+  continuous_invFun := e.continuous_toFun
+
+@[simp]
+theorem symm_toAffineEquiv (e : P₁ ≃ᵃL[k] P₂) : e.toAffineEquiv.symm = e.symm.toAffineEquiv :=
+  rfl
+
+@[simp]
+theorem symm_toEquiv (e : P₁ ≃ᵃL[k] P₂) : e.toEquiv.symm = e.symm.toEquiv := rfl
+
+-- custom simps projections??
+
+-- bijectivity, range_eq??
+
```

Comment:
You set up custom simps projections above, right?
```suggestion
```
Feel free to add the other missing lemmas but I don't think we should leave comments like this lying around, not least because it's too easy for them to get out of sync with the code.